packages/wallets/src/components/UIProvider/UIProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useContext, useState } from 'react';
+
+type Region = 'EU' | 'Non-EU';
+
+type TUIState = {
+    region: Region;
+};
+
+type TUIContext = {
```

Comment:
whats UI provider? it seems that it just holds extra state? is it effectively "AdditionalStateProvider"?

Reply:
Because we have regions regulations in traders hub. Currently we use store for it. Its basically changes ui based on selected region. Didnt wanna stuff modal provider with more stuff

Reply:
We plan to put themes and other ui related states here in the future

Reply:
:D :D completely agree with not stuffing modal provider with more stuff :D 
isn't it effectively creating an equivalent of smaller store with limited functionality? as effectively it does have interface to send action (update state), some logic to update state (as stores do) and basically feels like store. 

where does the "region" come from? is it dynamic / can it change during using session, or is it loaded once and thats it?

Reply:
I understand your point. But its used globally throughout the package. Its a switcher that user can switch from Non-EU to EU and vice versa. Based on that we show SVG or Maltainvest content. The user can change it as much as they want

Reply:
There are many edge cases that we hide stuff for EU content or the other. Thats why we need the setter and getter wherever possible

Reply:
okay, now I undestand the case, sure, makes sense,

Reply:
actually, would be cool to make it somewhat specific, e.g. "RegionProvider", to make sure, that its not gonna just became a bag for all the random props and random state out there. Basically follow DDD.

Reply:
They are planning to remove this regulation switcher in the future, so for the time being just wanna name this provider to be generic so its not specific

Reply:
if they want to remove it, thats even bigger reson to make it very narrow and specific - otherwise, its gonna contains tons of stuff unrelated to it, gonna be bag for everything, gonna became tightly coupled with all random functions here and there and then its gonna be quite impossible to remove it,